digitalaidseattle/games-for-love
# Mayk/popup2

## Description

This PR includes the following:

1. Refactoring of the `components` folder. (Changing the folder structure to separate `popup.tsx` and card offers advantages such as modularity, easier maintenance, improved readability, easier testing, and enhanced collaboration efficiency.)
2. Usage of `HospitalCard.tsx` inside `GFLPopup.tsx`.
3. Bug fix in the image section of `hospitalInfoService`.
4. Hospitals actively funding are marked in green on the map, while closed hospitals are marked in red.
5. Update the Popup UI to match the Figma design.

<img width="1104" alt="Screenshot 2024-08-07 at 2 59 52 PM" src="https://github.com/user-attachments/assets/35e22c1f-a6aa-4ee3-9f81-87334942dabe">


 ****

**Bug fix in the image section of `hospitalInfoService`**
![Screenshot 2024-08-07 at 3 36 29 PM](https://github.com/user-attachments/assets/a210cff1-52e2-45e9-a68c-b1f7bbf89872)




 ****

**Hospitals actively funding are marked in green on the map, while closed hospitals are marked in red**

![Screenshot 2024-08-07 at 3 27 14 PM](https://github.com/user-attachments/assets/7121d1f1-1981-4e86-a7c5-99b42dae1ba3)


 ****

**Update the Popup UI to match the Figma design**

![Screenshot 2024-08-07 at 3 30 01 PM](https://github.com/user-attachments/assets/b6b081a7-d0d8-4805-84be-26b49c989a9c)

I wanted to find a close button in MUI that matches the Figma UI with a black 'X' on a white background. The closest option I found is this MUI button, but the 'X' part is transparent and can't be customized. Do you have any good ideas on how to achieve this?


## What type of PR is this? (check all applicable)


## Related Tickets & Documents

[1.1.1.1 Map display for past hospitals](https://projects.digitalaidseattle.org/project-tasks-details?recordId=recGkvwBZkBXk6slA)

[1.1.1.3 Display a dialog or thumbnail displaying details of the hospital on map](https://projects.digitalaidseattle.org/project-tasks-details?recordId=recgA8niD89fPTXDW)

- Related Issue #
- Closes #

## QA Instructions, Screenshots, Recordings

To test these changes:
- Hospitals **actively funding** are marked in **green** on the map, while **closed** hospitals are marked in **red**. 
- When a marker (pin) is clicked, the corresponding popup should appear. 
- When the close button in the popup is clicked, the popup should close. 
- The information inside the popup should correctly display the mock data. 
- The **'Learn more'** and **'Donate'** buttons are not functional at this time, which is expected.
